File: pubnub-gson/src/main/java/com/pubnub/internal/endpoints/channel_groups/AllChannelsChannelGroupImpl.java
package com.pubnub.internal.endpoints.channel_groups;

import com.pubnub.api.PubNubException;
import com.pubnub.api.builder.PubNubErrorBuilder;
import com.pubnub.api.endpoints.channel_groups.AllChannelsChannelGroup;
import com.pubnub.api.endpoints.remoteaction.ExtendedRemoteAction;
import com.pubnub.api.models.consumer.channel_group.PNChannelGroupsAllChannelsResult;
import com.pubnub.internal.PubNubCore;
import com.pubnub.internal.endpoints.DelegatingEndpoint;
import lombok.Setter;
import lombok.experimental.Accessors;

@Accessors(chain = true, fluent = true)
public class AllChannelsChannelGroupImpl extends DelegatingEndpoint<PNChannelGroupsAllChannelsResult> implements AllChannelsChannelGroup {
    @Setter
    private String channelGroup;

    public AllChannelsChannelGroupImpl(PubNubCore pubnub) {
        super(pubnub);
    }

    @Override
    protected ExtendedRemoteAction<PNChannelGroupsAllChannelsResult> createAction() {
        return pubnub.listChannelsForChannelGroup(channelGroup);
    }

    @Override
    protected void validateParams() throws PubNubException {
        if (channelGroup == null) {
            throw new PubNubException(PubNubErrorBuilder.PNERROBJ_GROUP_MISSING);
        }
    }
}

File: pubnub-kotlin/pubnub-kotlin-impl/src/main/kotlin/com/pubnub/internal/endpoints/channel_groups/AllChannelsChannelGroupImpl.kt
package com.pubnub.internal.endpoints.channel_groups

import com.pubnub.api.endpoints.channel_groups.AllChannelsChannelGroup
import com.pubnub.api.models.consumer.channel_group.PNChannelGroupsAllChannelsResult
import com.pubnub.internal.EndpointImpl
import com.pubnub.internal.PubNubImpl

/**
 * @see [PubNubImpl.listChannelsForChannelGroup]
 */
class AllChannelsChannelGroupImpl internal constructor(allChannelsChannelGroup: AllChannelsChannelGroupInterface) :
    AllChannelsChannelGroupInterface by allChannelsChannelGroup,
    AllChannelsChannelGroup,
    EndpointImpl<PNChannelGroupsAllChannelsResult>(allChannelsChannelGroup)
